### PR TITLE
[WOR-1346] Wording changes related to workspace access (auth domains, Azure group-constraints)

### PR DESCRIPTION
**Pull request by cahrens**
### Jira Ticket: https://broadworkbench.atlassian.net/browse/WOR-1346

Introduces the following changes related to when a user does not have access to a workspace.

1. If the user attempts to directly access the Workspace Dashboard (through the URL), the text of the no access message has been generalized. We do not know in this case whether the workspace even exists, so we cannot differentiate between Azure and GCP.

Before:

image

After:

image

2. In the Workspaces list, if the user does not have access due to an auth domain (GCP) or group policy (Azure) the workspace will be in the list, but clicking it will show a dialog instead of going to the dashboard. The language has been generalized (and "click" is not a good thing to use as it presumes using a mouse).

Before:

image

After:

image

3. The dialog that is opened will remain unchanged for GCP workspace, but Azure workspaces will show this one instead:

image

The link goes to: https://support.terra.bio/hc/en-us/articles/19124069598235

For reference, this is the GCP version of the dialog:

image

### CODE REVIEW

**Comment by bethsheets** [comment]
@cahrens for #1, should we also include the troubleshooting tip "or your linked identity may have expired"?

**Comment by cahrens** [comment]
> @cahrens for #1, should we also include the troubleshooting tip "or your linked identity may have expired"?

@bethsheets I thought it intentionally wasn't in No. 1 because that is supposed to be generic for GCP and Azure. The article seemed to be very specific about NIH-access, which comes into play for Anvil… I was assuming it was more aimed at Azure (but perhaps I was wrong!).

What was the intention behind the "linked identity may have expired", and what would you like?

File: src/libs/ajax.js [hunk 1]
```diff
@@ -564,11 +564,6 @@ const Workspaces = (signal) => ({
         return res.json();
       },
 
-      accessInstructions: async () => {
```
Comment (cahrens) [inline src/libs/ajax.js:567]: This method was being called, but the results were never used.

File: src/pages/workspaces/workspace/RequestAccessModal.ts [hunk 2]
```diff
@@ -1,47 +1,71 @@
...
-  const fetchGroups = withErrorReporting('Error loading groups')(async () => {
-    setGroups(await Groups.list());
-  });
+interface GcpRequestAccessModalProps {
+  onDismiss: () => void;
+  workspaceInfo: GoogleWorkspaceInfo;
+}
 
-  const fetchAccessInstructions = withErrorReporting('Error loading instructions')(async () => {
-    setAccessInstructions(await Workspaces.workspace(workspace.workspace.namespace, workspace.workspace.name).accessInstructions());
```
Comment (cahrens) [inline src/pages/workspaces/workspace/RequestAccessModal.ts:25]: Access instructions were not actually being used.

**Review by nawatts — APPROVED** [review]

**Review by marctalbott — APPROVED** [review]

**Comment by cahrens** [comment]
Integration test failures are unrelated and have been reported to various teams.

**Pull request merged**